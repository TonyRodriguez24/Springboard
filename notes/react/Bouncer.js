const Bouncer = (props) => {
    let reply;
    if (props.age < 18) {
        reply = 'sorry kid can not come in'
    } else if (props.age < 21){
        reply = 'you can come in but no drinking'
    }
    else {
        reply =
            <p>
                Come in you can drink
                <img src = 'randomimage'/>
            </p>
    }
    return (
        <div>
            <p>
                <b>Bouncer:</b>  How old are you? 
            </p>
            <p>
                <b>You:</b> I am {props.age} years old
            </p>
            <p>
                <b>Bouncer: {reply}</b>
            </p>
        </div>
    )
}