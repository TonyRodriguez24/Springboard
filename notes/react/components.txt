building blocks of react
Pieces of UI and view logic
- using functions
    functions that know how to return HTML to render
    