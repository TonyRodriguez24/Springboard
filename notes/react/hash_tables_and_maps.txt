x Maps
    - abstract data type for mapping key value pairs
    - dictionary in python

Hash Maps
- mostly constant time
- choosing array size large enough to minimize collisions
- even with that collisions will happen

chaining
- sharing one bucket

open addressing
- linear probing 

both can get close to linear time in the worst case

x Set is just a Map without Values

JS
- sets
- objects
- Maps


python
- dictionary
- sets
- frozen set

all built on maps