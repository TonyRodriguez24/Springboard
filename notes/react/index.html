<!doctype html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Shibas</title>
  </head>
  <body>
    <div id = 'root'>
    
    </div>






    <script
      src="https://unpkg.com/react@18/umd/react.development.js"
      crossorigin></script>
    <script
      src="https://unpkg.com/react-dom@18/umd/react-dom.development.js"
      crossorigin></script>
    <script src="https://unpkg.com/@babel/standalone/babel.min.js"></script>
    <script src = 'random_num.js' type = 'text/jsx'></script>
    <script src = 'Animal.js' type = 'text/jsx'></script>
    <script src = 'RandomChoice.js' type = 'text/jsx'></script>
    <script src = 'Bouncer.js' type = 'text/jsx'></script>
    <script src = 'TodoList.js' type = 'text/jsx'></script>
    <script src = 'RandomNumberRange.js' type = 'text/jsx'></script>
    <script src = 'Alert.js' type = 'text/jsx'></script>
    <script src = 'App.js' type = 'text/jsx'></script>
  </body>
</html>
