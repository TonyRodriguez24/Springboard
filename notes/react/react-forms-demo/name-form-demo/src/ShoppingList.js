import React, { useState } from "react";
import Item from "./Item";
import NewItemForm from "./NewItemForm";
import { v4 as uuidv4 } from "uuid";

const ShoppingList = () => {
    const INITIAL_STATE = [
        { id: uuidv4(), name: 'peanut butter', qty: 2 },
        { id: uuidv4(), name: 'jelly', qty: 3 },
        { id: uuidv4(), name: 'milk', qty: 1 }
    ]
    const [items, setItems] = useState(INITIAL_STATE);

    const addItem = (newItem) => {
        setItems(items => [...items, { ...newItem, id: uuidv4()}])
    }
    return (
        <div>
            <h3>ShoppingList</h3>
            <NewItemForm addItem={addItem} />
            <div>
                {items.map(({name, qty, id}) => <Item name={name} qty={qty} key={id} />)}
            </div>
        </div>
    )
}

export default ShoppingList;
