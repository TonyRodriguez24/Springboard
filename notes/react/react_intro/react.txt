x Looping
    - using map the array method
    - {props.items.map}

!! React !!
will see map method everywhere and ternary operators
- map makes a new array, leaving original array unaffected
often use the implicit return (

) 

instead of 

{
    return ...
}

x Components
building blocks of react
Pieces of UI and view logic
- using functions
    functions that know how to return HTML to render
    

x Key props
- 