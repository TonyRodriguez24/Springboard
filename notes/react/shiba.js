



const App = () => {
    return (
        <div>
            <RandomNum />
            <RandomNum />
            
        </div>
    )
}



const Shiba = () => {
    return <img src= 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcS04F5bzlb41xlNPBathGI1EQheorogVLZQHA&s'></img>
}

ReactDOM.render(<App />, document.getElementById('root'));