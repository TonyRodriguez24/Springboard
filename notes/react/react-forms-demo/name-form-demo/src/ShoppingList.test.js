import { render } from "@testing-library/react";
import React from 'react';
import ShoppingList from './ShoppingList';

it('renders without crashing', function () {
    render(<ShoppingList />)
});

it("matches snapshot", function () {
    const { asFragment } = render(<ShoppingList />);
    expect(asFragment()).toMatchSnapshot();
});

it("should add new item", function () {
    const { queryByText } = render(<ShoppingList />);
    expect(queryByText('Name: Chocolate Milk')).not.toBeInTheDocument();
})