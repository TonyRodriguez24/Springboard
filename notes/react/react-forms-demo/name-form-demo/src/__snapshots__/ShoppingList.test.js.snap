// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      ShoppingList
    </h3>
    <form>
      <label
        for="name"
      >
        Product
      </label>
      <input
        id="name"
        name="name"
        placeholder="item name"
        type="text"
        value=""
      />
      <label
        for="qty"
      >
        Quantity 
      </label>
      <input
        id="qty"
        max="10"
        min="1"
        name="qty"
        type="range"
        value=""
      />
      <button>
        Add Item
      </button>
    </form>
    <div>
      <ul>
        <li>
          Name: peanut butter
        </li>
        <li>
          Quantity: 2
        </li>
      </ul>
      <ul>
        <li>
          Name: jelly
        </li>
        <li>
          Quantity: 3
        </li>
      </ul>
      <ul>
        <li>
          Name: milk
        </li>
        <li>
          Quantity: 1
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
